### PR TITLE
Handler settings types, parsing and validation

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**

### CODE REVIEW

File: handlersettings.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/Azure/azure-docker-extension/pkg/vmextension"
+	"github.com/pkg/errors"
+)
+
+var (
+	errStoragePartialCredentials = errors.New("'storageAccountName' and 'storageAccountKey' must be both specified")
```
Comment (boumenot) [inline handlersettings.go]: The phrase sounds strange due to the placement of both.  I would either drop it, or move it to beginning of the sentence.

File: handlersettings.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	errCmdTooMany                = errors.New("'commandToExecute' was specified both in public and protected settings; it should be specified only once")
```
Comment (boumenot) [inline handlersettings.go]: nit: should => must

**Comment by boumenot** [comment]
LGTM.

**Comment by ahmetalpbalkan** [comment]
addressed CR comments, merging. Thanks.

**Pull request merged**